### PR TITLE
Add ignore_index to Accuracy metric

### LINKED ISSUE #61: Add ignore_index option to Accuracy metric
## 🚀 Feature

It would be nice if `Accuracy` metric accepted `ignore_index` as it works for `StatsScore` or `torch.nn.CrossEntropyLoss`

### PR DESCRIPTION
**Pull request by avinashsai**
Added ignore_index to Accuracy metric. Fixes #61

### CODE REVIEW

**Comment by avinashsai** [comment]
@SkafteNicki Please, review this

**Comment by SkafteNicki** [comment]
Hi @avinashsai,
Code is looking great, but could you add some kind of test showing it is working as intended?

**Comment by avinashsai** [comment]
I haven't added testcases. But I have checked externally with all cases. Should I add test cases in test_accuracy.py?

**Comment by SkafteNicki** [comment]
> added

Yes please do so :]

File: torchmetrics/classification/accuracy.py [hunk 1]
```diff
@@ -105,6 +109,7 @@ def __init__(
         self,
         threshold: float = 0.5,
         top_k: Optional[int] = None,
+        ignore_index: Optional[int] = None,
```
Comment (tchaton) [inline torchmetrics/classification/accuracy.py:112]: Could we support `ignore_indexes` directly as `Optional[Union[List, int]]` ?

File: torchmetrics/functional/classification/accuracy.py [hunk 2]
```diff
@@ -34,6 +36,17 @@ def _accuracy_update(
...
 
+    if ignore_index is not None and not 0 <= ignore_index < preds.shape[1]:
+        raise ValueError(f"The `ignore_index` {ignore_index} is not valid for inputs with {preds.shape[0]} classes")
+
+    if ignore_index is not None and preds.shape[1] == 1:
+        raise ValueError("You can not use `ignore_index` with binary data.")
+
+    # Delete what is in ignore_index, if applicable (and classes don't matter):
+    if ignore_index is not None:
+        preds = _del_column(preds, ignore_index)
```
Comment (tchaton) [inline torchmetrics/functional/classification/accuracy.py:47]: Shoudn't we filter on `target` and apply the mask on target ?

**Review by tchaton — COMMENTED** [review]
Please, add test with different number of ignore_index in preds, targets.

File: torchmetrics/classification/accuracy.py [hunk 3]
```diff
@@ -142,7 +148,8 @@ def update(self, preds: Tensor, target: Tensor):
         """
 
         correct, total = _accuracy_update(
-            preds, target, threshold=self.threshold, top_k=self.top_k, subset_accuracy=self.subset_accuracy
+            preds, target, threshold=self.threshold, top_k=self.top_k, ignore_index=self.ignore_index,
+            subset_accuracy=self.subset_accuracy
```
Comment (Borda) [inline torchmetrics/classification/accuracy.py:152]:
```suggestion
            subset_accuracy=self.subset_accuracy,
```
and use yapf

**Review by Borda — CHANGES_REQUESTED** [review]
add test for usage and raised new error

File: torchmetrics/functional/classification/accuracy.py (hunk 2 above)
Comment (avinashsai) [reply torchmetrics/functional/classification/accuracy.py:47]: This is how ignore_index is done in precision.  That's why I didn't want to add new way.

**Comment by avinashsai** [comment]
@SkafteNicki  I dont know why tests are failing.

**Comment by Borda** [comment]
> @SkafteNicki I dont know why tests are failing.

it seems that the outputs do not meet expectations
https://github.com/PyTorchLightning/metrics/pull/155/checks?check_run_id=2331922213

**Comment by Borda** [comment]
@avinashsai mind check the failing tests?
